### ISSUE TITLE
[Bug] aptos move test doesn't work well with module addresses

### ISSUE THREAD

**Issue by movekevin** [aptos-labs/aptos-core#7228]
For a module defi::module_name, if the Move.toml has defi="_" under [addresses], aptos move publish works but aptos move test doesn't. For aptos move test to work, an explicit named address needs to be passed or defi needs to be set to some random address (e.g. 0xcafe) in Move.toml.

Adding defi under [dev-addresses] would lead to https://github.com/aptos-labs/aptos-core/issues/6264

**Comment by borispovod** [comment]
Hi,

Do you have any timeline for it?

It's quite strange that two packages can't have the same named addresses. We recently also encountered some issues with it, for example:

“Error”: “Move compilation failed: Unable to resolve packages for package ‘migrate’: Resolving named addresses for dependency ‘XX1’ in package ‘migrate’: Named address ‘XXX’ in dependency ‘XX1’ is already set to ‘0x6...’ but was then reassigned to ‘0xb2...’”

**Comment by gregnazario** [comment]
I've added the ability to use `dev-addresses` in https://github.com/aptos-labs/aptos-core/pull/8789

This should then let for CI or anything else to just put `--dev` and use those instead of the `addresses` field

**Comment by gregnazario** [comment]
> Hi,
> 
> Do you have any timeline for it?
> 
> It's quite strange that two packages can't have the same named addresses. We recently also encountered some issues with it, for example:
> 
> ```
> “Error”: “Move compilation failed: Unable to resolve packages for package ‘migrate’: Resolving named addresses for dependency ‘XX1’ in package ‘migrate’: Named address ‘XXX’ in dependency ‘XX1’ is already set to ‘0x6...’ but was then reassigned to ‘0xb2...’”
> ```

Having the same name is harder since they conflict, and I think it's separate of here, but @movekevin can probably have more insight on this

**Comment by banool** [comment]
Relevant issues regarding the named address collision:
- https://github.com/aptos-labs/aptos-core/issues/6464
- https://github.com/move-language/move/issues/867

**Comment by 0xjinn** [comment]
So I noticed that when you have a dev-addresses e.g `test=0x123` in toml file and you do `aptos move test --named-addresses test=0x123`, this will result in an error similar to:

```
{
  "Error": "Unexpected error: Failed to run tests: Unable to resolve packages for package 'cli_e2e_tests': Unable to resolve named address 'addr' inpackage 'cli_e2e_tests' when resolving dependencies in dev mode: Attempted to assign a different value '0x12345' to an a already-assigned named address '0xe6fc96a9464ef0a85fd7146fdbae0a48c5930e44d5552ebca90ad0d0a2c7636c'"
}
```

I wonder, can I update the unify function here to always override if there is additional named addresses that has the same name? (Not sure if elsewhere using the additional named address other than CLI)
https://github.com/aptos-labs/aptos-core/blob/main/third_party/move/tools/move-package/src/resolution/resolution_graph.rs#L814-L817

**Comment by alinafoster4583** [comment]
Yes, there’s some kind of givеаway going on. I got some Ethereum there yesterday for different wallets. It’s unlikely they’ll take it back from me, but it’s a pity I didn’t have more wallets and couldn’t get more. But maybe it’s for the best.

<img width="369" alt="Image" src="https://github.com/user-attachments/assets/6f428b2a-76c4-4baf-be67-42bf6f8e4bfd" />

<img width="373" alt="Image" src="https://github.com/user-attachments/assets/b2218904-9f6a-4ea9-98cd-713c5b7d8e4a" />
